Core/src/IfcGeometryConverter/PlacementConverter.h

```diff
@@ -35,1657 +35,1664 @@
+                        // END_ENTITY;
+                        // **************************************************************************************************************************
+
+                        // RelativePlacement				
+                        // TYPE IfcAxis2Placement = SELECT (
+                        //	IfcAxis2Placement2D,
+                        //	IfcAxis2Placement3D);
+                        // END_TYPE;
+                        convertRelativePlacement(object_placement_matrix, local_placement);
+                        carve::math::Matrix relative_placement(carve::math::Matrix::IDENT());
+                        convertIfcLocalPlacementRelationPoint(alreadyApplied, local_placement, relative_placement);
+                        object_placement_matrix = relative_placement * object_placement_matrix;
+
+                    }
+
+                    void convertRelativePlacementOriginInIfcLinearPlacement(std::vector<std::shared_ptr<typename IfcEntityTypesT::IfcObjectPlacement>>& alreadyApplied, carve::math::Matrix& object_placement_matrix, std::shared_ptr<typename IfcEntityTypesT::IfcLinearPlacement> linear_placement);
+
+                    void checkLinearPlacementAgainstAbsolutePlacement(carve::math::Matrix& matrix, std::shared_ptr<typename IfcEntityTypesT::IfcLinearPlacement> linear_placement)
+                    {
+                        // ***********************************************************
+                        // check with the provided CartesianPosition
+                        // CartesianPosition OPTIONAL
+                        carve::math::Matrix absolute_placement(carve::math::Matrix::IDENT());
+                        if(linear_placement->CartesianPosition) {
+                            convertIfcAxis2Placement3D(linear_placement->CartesianPosition.get().lock(), absolute_placement);
+                            if(absolute_placement != matrix) {
+                                BLUE_LOG(trace) << linear_placement->getErrorLog() << "Absolute placemet and the calculated linear placement do not agree";
+                            }
+                        }
+                    }
+
+                    carve::geom::vector<3> getOffsetFromCurve(double length_factor, std::shared_ptr<typename IfcEntityTypesT::IfcDistanceExpression> distExpr)
+                    {
+                        return carve::geom::VECTOR(
+                            distExpr->OffsetLongitudinal.value_or(0.0) * length_factor,
+                            distExpr->OffsetLateral.value_or(0.0) * length_factor,
+                            distExpr->OffsetVertical.value_or(0.0) * length_factor
+                        );
+                    }
+
+                    std::tuple< carve::geom::vector<3>, carve::geom::vector<3>> calculatePositionOnAndDirectionOfBaseCurve(std::shared_ptr<typename IfcEntityTypesT::IfcLinearPlacement> linear_placement, double length_factor)

```

Comment:
Doxycomment missing. Also, this is just a wrapper for convertBoundedCurveDistAlongToPoint3D. Why not just change the signature of that function? Shouldn't the return tuple be returned as reference to save copying actions?